### PR TITLE
Add inabox envelopes for friendly and safeframe.

### PR DESCRIPTION
**Pull request by zombifier**
For #23569.

Also remove the obsolete mode number from the inabox envelope.

MRAID support is somewhat more difficult, and will come later (or within the same PR).

### CODE REVIEW

**Comment by zombifier** [comment]
Hmm this probably shouldn't be merged just yet. I'm relying on content-type to check for text/html as to avoid pasting binary text into HTML files (in case a local test page uses relative path and also route the embedded resources through the envelope), but not all sites will set them. It might be better to just update the examples themselves.

File: build-system/routes/a4a-envelopes.js [hunk 1]
```diff
@@ -48,6 +49,79 @@ app.use('/inabox/', (req, res) => {
...
+// http://localhost:8000/inabox-friendly/proxy/s/www.washingtonpost.com/amphtml/news/post-politics/wp/2016/02/21/bernie-sanders-says-lower-turnout-contributed-to-his-nevada-loss-to-hillary-clinton/
+app.use('/inabox-friendly', (req, res) => {
+  let adUrl = req.url;
+  const templatePath = '/build-system/server-inabox-friendly-template.html';
+  const urlPrefix = getUrlPrefix(req);
+  adUrl = addQueryParam(adUrl, 'inabox', 1);
+  if (req.query.log) {
+    adUrl += '#log=' + req.query.log;
+  }
+  fs.readFileAsync(process.cwd() + templatePath, 'utf8').then(template => {
```
Comment (lannka) [inline build-system/routes/a4a-envelopes.js]: pls extract method to reduce code dup

File: build-system/server-inabox-friendly-template.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  </style>
+</head>
+<body>
+  <h1>In-a-box Friendly Envelope</h1>
+  <div>url: AD_URL</div>
+  <div>size: AD_WIDTHxAD_HEIGHT</div>
+
+  <div style="background: #eee; overflow: hidden;; height: OFFSET;">scroll down to see the ad</div>
+
+  <iframe
```
Comment (lannka) [inline build-system/server-inabox-friendly-template.html]: what about reusing server-inabox-template and make the whole `<iframe>` section dynamic?

**Comment by lannka** [comment]
you might have missed this comment: https://github.com/ampproject/amphtml/pull/23699/files#r311754120

what you think?

**Comment by zombifier** [comment]
Yeah I was thinking of ways to implement it. I'll ping you once it's ready.

File: build-system/server-inabox-friendly-template.html (hunk 2 above)
Comment (zombifier) [reply build-system/server-inabox-friendly-template.html]: Changed it.

File: build-system/routes/a4a-envelopes.js (hunk 1 above)
Comment (zombifier) [reply build-system/routes/a4a-envelopes.js]: Done.

**Comment by zombifier** [comment]
All envelopes should now use only one template, PTAL.

**Comment by lannka** [comment]
@zombifier one more thing, for friendly iframe, we don't need the ads-tag-integration.js

**Pull request merged**